lib/route.method.js

```diff
@@ -30,7 +31,10 @@ function createRouteInterceptor(routeMethod) {
         }
 
         function isAuthorizationRequired() {
-            return authorizationStrategy instanceof Function && routeMethod.authorizationRequired !== false;
+            var isMethodNotAllowed = methodNotAllowed === routeMethod.handlerFunc;
```

Comment:
This is confusing, partially because you're using two vars that seem nearly identical, and partially because one of them is being compared to routeMethod.handlerFunc, which seems insane. I understand that methodNotAllowed is a function- perhaps we need to rename it methodNotAllowedFunc?

This would allow us to use methodAllowed Or methodNotAllowed as a boolean, which lines up with the other booleans defined directly below in terms of syntax. 

Also, you don't have to use (), for these comparison, but you should because it'll make the results more obvious to someone glancing at the code. 

We want to prioritize ease of reading code over ease of writing it, since for any code you write, it is likely that much more time will be spent maintaining & reading it than you spent writing it.